### PR TITLE
Minor fixes

### PR DESCRIPTION
**Pull request by pedropombeiro**
[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)

### Summary

[comment]: # (Summarise the problem and how the pull request solves it)

This PR serves to offload some work from the Nix gradle/node2nix PR, to make that one as light as possible.

### Review notes

This PR:
- moves files which are no longer different between mobile and desktop to the root, so we have as few symlinks as possible.
- fixes the `GIT_ROOT` variable so that it works even if the script is invoked from within a folder with a different git project (I needed this for a Nix build).
- updates `react-native-dialogs` to 1.0.4, since 1.0.2 contains a duplicate type which caused a build error.

#### Platforms

- Android
- Linux

#### Areas that maybe impacted

Should have no impact on the app, unless `react-native-dialogs` 1.0.4 is somehow broken.

status: ready

### CODE REVIEW

File: scripts/gen-deps-hash.sh [hunk 1]
```diff
@@ -6,7 +6,7 @@
 # the combined versions of all the specified tools.
 ################################################################################
 
-GIT_ROOT=$(git rev-parse --show-toplevel)
+GIT_ROOT=$(cd "${BASH_SOURCE%/*}" && git rev-parse --show-toplevel)
```
Comment (jakubgs) [inline scripts/gen-deps-hash.sh:9]: I get it, but why exactly is this necessary?

Comment (jakubgs) [reply scripts/gen-deps-hash.sh:9]: Oh:
>if the script is invoked from within a folder with a different git project 

Sorry, didn't read the description =].

**Comment by yenda** [comment]
@PombeirP some of these changes are already in https://github.com/status-im/status-react/pull/8334 could you have a look at it please and documents how things will work now with nix, do we have to change the nix node modules file as well now when changing deps?

**Comment by pedropombeiro** [comment]
> @PombeirP some of these changes are already in #8334 could you have a look at it please and documents how things will work now with nix, do we have to change the nix node modules file as well now when changing deps?

I've added a post in Discuss to shed some light into what is changing: https://discuss.status.im/t/the-road-to-nix-a-functional-package-manager-to-rule-them-all/1049/114?u=pedro

**Review by rasom — APPROVED** [review]

**Comment by pedropombeiro** [comment]
@rasom do you know what happened with the builds? It's mentioning status-modules, but as far as I can tell, my branch isn't rebased on latest develop which contains that.

File: ci/utils.groovy [hunk 2]
```diff
@@ -1,11 +1,7 @@
 nix = load 'ci/nix.groovy'
 
 def getVersion(type = null) {
```
Comment (jakubgs) [inline ci/utils.groovy]: Then I guess we should drop the `type` argument and remove it from all the `getVersion()` calls.

File: nix/desktop/realm-node/generate-nix.sh [hunk 3]
```diff
@@ -20,12 +21,21 @@ cat << EOF > $input
 ]
 EOF
 
+# Specify the package.json file containing the build dependencies to install
+cat << EOF > $supplement_input
+[
+  "node-pre-gyp"
+]
+EOF
```
Comment (jakubgs) [inline nix/desktop/realm-node/generate-nix.sh:29]: What's the need for this multi-line syntax?

**Review by jakubgs — APPROVED** [review]
Love how much stuff this removes.

File: nix/desktop/realm-node/generate-nix.sh (hunk 3 above)
Comment (pedropombeiro) [reply nix/desktop/realm-node/generate-nix.sh:29]: this file might change often and include more modules, I just don't want to keep changing syntax as we add and remove modules.

File: ci/utils.groovy (hunk 2 above)
Comment (pedropombeiro) [reply ci/utils.groovy]: You're right, I missed that, fixing.

File: nix/desktop/realm-node/generate-nix.sh (hunk 3 above)
Comment (jakubgs) [reply nix/desktop/realm-node/generate-nix.sh:29]: fair

**Pull request merged**